dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeDestroyAPIImpl.java

```diff
@@ -68,32 +130,53 @@ public void relocateContentletsForDeletion(final ContentType source, final Conte
             );
         }
 
-        final DotConnect dotConnect = new DotConnect();
+        final String tempTableName = String.format("contentlets_to_be_updated_%d", System.currentTimeMillis());
 
-        final String selectContentlets = "select c.inode from contentlet c where c.structure_inode =  ? \n";
-        dotConnect.setSQL(selectContentlets).addParam(source.inode());
+        //The quickest way to relocate contentlets is to create a temp table with the inodes of the contentlets to be relocated
+        final String dumpContentletToUpdate = dumpTableScript(tempTableName, source);
 
-        String updateContentlet = String.format(
-                "update contentlet c set structure_inode = ?, \n" +
-                        " contentlet_as_json = jsonb_set(contentlet_as_json,'{contentType}', '\"%s\"'::jsonb, false)  \n" +
-                        " where c.structure_inode =  ? ", target.inode()
-        );
+        Try.of(()->new DotConnect().executeStatement(dumpContentletToUpdate)).onFailure(e->{
+            Logger.error(this.getClass(), "Error trying to create temp table to update contentlets", e);
+            throw new DotStateException(e);
+        });
+
+        final int allCount = new DotConnect().setSQL(String.format("select count(*) as count from %s", tempTableName)).getInt("count");
+
+        int limit = relocateLimitProp.get();
+        int from = 0;
+        int to = limit;
+
+        int times = allCount <= limit ?  1 : (int)Math.ceil(allCount / (float)limit);
+
+        Logger.debug(this.getClass(),
+                String.format("Relocating (%d) contentlets from (%s) to (%s) in (%d) transactions..", allCount, source.name(), target.name(), times));
+
+        for (int i = 0; i < times; i++) {
+
+            Logger.debug(this.getClass(),
+                    String.format("Relocated from (%d) to (%d) of (%d) contentlets..", from, to, allCount));
+
+            final String updateStatement = updateStatement(target, tempTableName, from, to);
+            final DotConnect updateConnect = new DotConnect().setSQL(updateStatement);
+            updateConnect.loadResults();
+
+            from += limit;
+            to += limit;
+
+            if(i % 100 == 0){
+                Logger.info(this.getClass(),
```

Comment:
would this logger.info generate too much noise?

Reply:
the idea is that it does not always render. 
I decided to module by 100 so it wouldn't render on every iteration. but it also helps to know whats going on.